### PR TITLE
Added some simple JMH benchmarks

### PR DESCRIPTION
**Pull request by dlew**
I don't know that this is representative, but it gives us a place to start for #17.

### CODE REVIEW

File: rxdogtag/build.gradle [hunk 1]
```diff
@@ -26,4 +27,13 @@ dependencies {
   testCompile deps.test.truth
 }
 
+jmh {
+  jmhVersion = deps.versions.jmhLib
+
+  if (project.hasProperty("jmh")) {
```
Comment (ZacSweers) [inline rxdogtag/build.gradle]: For the sake of simplicity since it's small right now, let's remove this

**Comment by ZacSweers** [comment]
Do you want to paste your current results here for posterity? When we have an android benchmark in too I'll put together a wiki page

File: rxdogtag/build.gradle (hunk 1 above)
Comment (dlew) [reply rxdogtag/build.gradle]: I'm fine removing this, though I found it to be a very useful trick when experimenting with different setups and seeing how it affected runs. :)

Comment (ZacSweers) [reply rxdogtag/build.gradle]: Hmmm, fair point. Let's keep it then but maybe add a comment above it with instructions on how to run

**Comment by dlew** [comment]
For the record, here's what I'm currently getting on running this specific branch + benchmarks: [benchmark.txt](https://github.com/uber/RxDogTag/files/3189701/benchmark.txt)

**Comment by ZacSweers** [comment]
Pasting to save others a download

```
Benchmark                                 (installRxDogTag)  (times)  (useRxDogTag)   Mode  Cnt          Score        Error  Units
EntireProcessPerf.flowable                            false        1            N/A  thrpt   25   33466069.460 ± 109373.380  ops/s
EntireProcessPerf.flowable                            false     1000            N/A  thrpt   25     306413.876 ±    421.045  ops/s
EntireProcessPerf.flowable                            false  1000000            N/A  thrpt   25        255.349 ±      0.537  ops/s
EntireProcessPerf.flowable                             true        1            N/A  thrpt   25     941855.142 ±   5608.411  ops/s
EntireProcessPerf.flowable                             true     1000            N/A  thrpt   25      42690.908 ±     63.684  ops/s
EntireProcessPerf.flowable                             true  1000000            N/A  thrpt   25         51.087 ±      0.102  ops/s
EntireProcessPerf.observable                          false        1            N/A  thrpt   25   44497040.465 ±  47773.975  ops/s
EntireProcessPerf.observable                          false     1000            N/A  thrpt   25     306785.991 ±    492.110  ops/s
EntireProcessPerf.observable                          false  1000000            N/A  thrpt   25        297.383 ±      0.455  ops/s
EntireProcessPerf.observable                           true        1            N/A  thrpt   25     946489.198 ±   5947.795  ops/s
EntireProcessPerf.observable                           true     1000            N/A  thrpt   25      40846.515 ±     68.068  ops/s
EntireProcessPerf.observable                           true  1000000            N/A  thrpt   25         42.935 ±      0.338  ops/s
EventHandlingPerf.onNext                                N/A      N/A          false  thrpt   25  317514566.096 ± 488408.006  ops/s
EventHandlingPerf.onNext                                N/A      N/A           true  thrpt   25   52701769.084 ±  89595.047  ops/s
SubscriptionPerf.subscribeWithOnError                 false      N/A            N/A  thrpt   25   73568366.134 ± 149037.817  ops/s
SubscriptionPerf.subscribeWithOnError                  true      N/A            N/A  thrpt   25   51601468.150 ± 119841.519  ops/s
SubscriptionPerf.subscribeWithoutOnError              false      N/A            N/A  thrpt   25   76331264.343 ± 118066.988  ops/s
SubscriptionPerf.subscribeWithoutOnError               true      N/A            N/A  thrpt   25     986172.439 ±   4700.472  ops/s
```

**Comment by ZacSweers** [comment]
Thanks! I've started an Android version in #27 as well. Once that's in we'll look at unifying the cases a bit. Would be curious for your thoughts on those tests as well!

**Pull request merged**